File: internal/topics/topics.go
// Package topics names every SSE topic Hearth publishes on. It is the SSE
// contract shared with the frontend (web/src/topics.ts mirrors it): each
// widget publishes on its own ID, platform code publishes on concept
// topics, and a payload of "changed" means "re-fetch". Adding a topic here
// means adding it to the frontend mirror too.
package topics

// Widget topics — one per registered widget, equal to the widget's ID.
const (
	Clock     = "clock"
	Calendar  = "calendar"
	Chores    = "chores"
	Grocery   = "grocery"
	Meds      = "meds"
	Weather   = "weather"
	Guestbook = "guestbook"
	MealPlan  = "mealplan"
	Sports    = "sports"
)

// Platform topics — concepts owned by the server, not any single widget.
const (
	Views    = "views"
	Profiles = "profiles"
	Night    = "night"
	Guest    = "guest"
)

File: web/src/topics.ts
// Frontend mirror of the SSE topic contract (Go: internal/topics). Widgets
// publish on their own slug — widget.Base ties slug and topic together — and
// platform features (views, profiles, guest, night) have their own topics.
// The payload "changed" simply means "re-fetch whatever you show".
export const TOPICS = {
  clock: "clock",
  calendar: "calendar",
  chores: "chores",
  grocery: "grocery",
  meds: "meds",
  weather: "weather",
  guestbook: "guestbook",
  mealplan: "mealplan",
  sports: "sports",
  news: "news",
  views: "views",
  profiles: "profiles",
  night: "night",
  guest: "guest",
} as const;
